File: LeetCode/129. Sum Root to Leaf Numbers/Solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode], num: int = 0) -> int:
        if not root: return 0
        num = num * 10 + root.val
        if not root.left and not root.right: return num
        return self.sumNumbers(root.left, num) + self.sumNumbers(root.right, num)
        

File: LeetCode/129. Sum Root to Leaf Numbers/Solution.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int SumNumbers(TreeNode root) {
        return SumNumbers(root, 0);
    }

    private int SumNumbers(TreeNode root, int num) {
        if(root == null) return 0;
        num *= 10;

        if(root.left == null && root.right == null) return num + root.val;
        else return SumNumbers(root.left, num + root.val) + SumNumbers(root.right, num + root.val);
    }
}